### PR TITLE
ENH: Add credentials argument to read_gbq and to_gbq.

### PR DESCRIPTION
**Pull request by tswast**
Deprecates private_key parameter.

Note: I didn't actually add a deprecation warning message yet because I don't know which version of pandas this change will get added to.

Towards #161

### CODE REVIEW

**Comment by tswast** [comment]
Travis build: https://travis-ci.org/tswast/pandas-gbq/builds/446894492

File: docs/source/howto/authentication.rst [hunk 1]
```diff
@@ -18,11 +18,38 @@ Create a service account key via the `service account key creation page
...
+
+* :func:`google.oauth2.service_account.Credentials.from_service_account_file`,
+    which accepts a file path to the JSON file.
+
+    .. code:: python
+
+        credentials = google.oauth2.service_account.Credentials.from_service_account_file(
+            'path/to/key.json',
+        )
+        df = pandas_gbq.read_gbq(sql, project_id="YOUR-PROJECT-ID", credentials=credentials)
```
Comment (max-sixty) [inline docs/source/howto/authentication.rst]: Should we swap the order of these? I don't use service accounts outside of GCP much; is it right that people use files far more often than the writing the full definition?

Comment (tswast) [reply docs/source/howto/authentication.rst]: You're right. Files are more common. Swapped.

**Comment by max-sixty** [comment]
🙌

**Pull request merged**